WikiWatershed/model-my-watershed
# Add DRWI Perimeter, limit Recent Weather to it

## Overview

Adds a DRWI Simple Perimeter boundary, and limits Recent Weather to it. Recent Weather is only available for 11 stations, but is needed for areas slightly beyond the DRB. This is shown in QGIS:

![image](https://user-images.githubusercontent.com/1430060/173145377-56c23e33-0bc4-4e1f-b40a-c1277066ce56.png)

where the red dots are the Recent Weather data stations, the dark pink is the DRB, and the light pink is the additional area covered under DRWI. The yellow is the buffered simplified boundary we use to check if an area of interest is within the DRWI or not.

See commit messages for details.

Connects #3547

### Demo

![image](https://user-images.githubusercontent.com/1430060/173145574-bb75965a-2188-43e2-97a1-c5776a9ff48f.png)

## Testing Instructions

* Check out this branch and `bundle.sh --debug`
* Go to http://localhost:8000 and log in (since weather simulations can only be loaded for logged in users)
* Pick a shape within the DRB (e.g. City of Philadelphia-Schuylkill River HUC-12)
* Model it with Watershed Multi-Year Model
* Add Changes to make a new scenario
* Click Weather Data to load the weather modal
* Open the Available Data dropdown
  - [x] Ensure that NASA NLDAS 2000-2019 is available
  - [x] Ensure that RCP 4.5 and RCP 8.5 are available
* Pick a shape outside the DRB but within the DRWI, e.g. Landing Creek HUC-12
* Repeat the above steps to get to the Available Data dropdown
  - [x] Ensure that NASA NLDAS 2000-2019 is available
  - [x] Ensure that RCP 4.5 and RCP 8.5 are not available
* Pick a shape outside the DRWI, e.g. Elklick Creek HUC-12
* Repeat the above steps to get to the Available Data dropdown
  - [x] Ensure that NASA NLDAS 2000-2019 is not available
  - [x] Ensure that RCP 4.5 and RCP 8.5 are not available
